core/src/com/unciv/models/ruleset/unique/Unique.kt

```diff
@@ -50,6 +51,16 @@ class Unique(val text: String, val sourceObjectType: UniqueTarget? = null, val s
     fun getModifiers(type: UniqueType) = modifiersMap[type] ?: emptyList()
     fun hasModifier(type: UniqueType) = modifiersMap.containsKey(type)
     fun isModifiedByGameSpeed() = hasModifier(UniqueType.ModifiedByGameSpeed)
+    fun isModifiedByGameProgress() = hasModifier(UniqueType.ModifiedByGameProgress)
+    fun modifyByGameProgress(civ: Civilization, modifyPercent: Float): Float {
```

Comment:
If you're going to make a modifier function, you can have it do the whole "unique.getModifiers(UniqueType.ModifiedByGameProgress).firstOrNull()!!.params[0].toFloat()" thing
Since 'unique' is already here, as 'this'

Second, naming conventions - since this doesn't *do* the modifications but instead provides the modifier, it should be getGameProgressModifier